Unable to run matlab example

**Describe the Bug**
env：Win10，webots 2019b and MATLAB 2017b. 
Using webots to open its own matlab example, click to run it , the MATLAB window is blocked, there is frozen, can only be  closed by windows' TaskManager .  I have set WEBOTS_HOME, WEBOTS_CONTROLLER_NAME and WEBOTS_VERSION parameters .            
Later, I came up with a solution: empty the content of launcher. m,  then run this example.  After pop up the MATLAB window  stop by error,  I copy the content of launcher. m into the matlab window and run it again. This Webots example can run now, but there are still errors, as if there is some libraries not loaded.

**Steps to Reproduce**
1. open webots2019b
2. load C:\Program Files\Webots\projects\languages\matlab\worlds\e-puck_matlab.wbt
3. run it
4. matlab frozen

**Expected behavior**
Should be the normal operation of the program

**Screenshots**
If applicable, add screenshots to help explain your problem.

**System**
 - win10 64bit Chinese version
 - Graphics Card: AMD Radeon R9 M370X

Env: webots 2020a rev2, matlab 2017b, 
I loaded C: \ program files \ webots \ projects \ languages \ matlab \ worlds \ e-pump_matlab.wbt file and 
Change the robot's controller to < extern > . then run the launcher. m file(could not run matlab.m file) in Matlab,some error msg displayed :

>> launcher
Warning: name does not exist or is not a directory: C:\Program Files\Webots\lib\matlab 
> In path (line 109)
  In addpath (line 94)
  In launcher (line 23) 
The MATLAB "MinGW-w64 C/C++ Compiler" addon is not installed, please install it from: https://fr.mathworks.com/matlabcentral/fileexchange/52848-matlab-support-for-mingw-w64-c-c-compiler
Waiting up to 5 seconds for C:\Users\xsun\AppData\Local\Temp\webots_matlab_lock to be deleted by another MATLAB instance.
Deleting C:\Users\xsun\AppData\Local\Temp\webots_matlab_lock...
Creating: C:\Users\xsun\AppData\Local\Temp\protofile_matlab_2017b_webots_R2020a-rev1.m
Load Library failed.
Error use loadlibrary
Unable to preprocess input file.
The output generated by the preprocessor is:   C:\Program Files\Webots\lib\controller\matlab\allincludes.h:8:61: fatal error: ../../include/controller/c/webots/accelerometer.h: No such file or directory
 #include "../../include/controller/c/webots/accelerometer.h"  
compilation terminated.
Error launcher (line 74)
      loadlibrary( ...

How can I do? Thanks.

--- a/docs/guide/using-matlab.md
+++ b/docs/guide/using-matlab.md
@@ -15,18 +15,16 @@ In addition, it becomes possible to reuse your existing MATLAB code directly in 
 In order to use MATLAB controllers in Webots, the MATLAB software must be installed (a MATLAB license is required).
 Webots {{ webots.version.full }} supports only 64bits MATLAB versions from 2015b.
 
-On Windows, the [MATLAB MinGW-w64 C/C++ Compiler](https://fr.mathworks.com/matlabcentral/fileexchange/52848-matlab-support-for-mingw-w64-c-c-compiler) needs to be installed in addition to MATLAB.
+On Windows and MATLAB versions prior to R2019b, the [MATLAB MinGW-w64 C/C++ Compiler](https://fr.mathworks.com/matlabcentral/fileexchange/52848-matlab-support-for-mingw-w64-c-c-compiler) needs to be installed in addition to MATLAB.
 
 Webots must be able to access the "matlab" executable (usually a script) in order to run controller m-files.
 Webots looks for the "matlab" executable in every directory of your *PATH* (or *Path* on Windows) environment variable.
 Note that this is similar to calling "matlab" from a terminal (or *Command Prompt* on Windows), therefore, if MATLAB can be started from a terminal then it can also be started from Webots.
 
 On Windows, the MATLAB installer will normally add MATLAB's bin directories to your *Path* environment variable, so usually, Webots will be able to locate MATLAB after a standard installation.
-However, in case it does not work, please make sure that your *Path* contains this directory (or something slightly different, according to your MATLAB version):
+However, in case it does not work, please make sure that your *Path* environment variable contains this directory (or something slightly different, according to your MATLAB version): `C:\Program Files\MATLAB\R2019b\bin`.
+With older versions of MATLAB (e.g., R0217b), your *Path* environment variable should also include: `C:\Program Files\MATLAB\R2017b\bin\win64`.
 
-```
-Path=C:\Program Files\MATLAB\R2018b\bin
-```
 
 On Linux, the MATLAB installer normally suggests to add a symlink to the "matlab" startup script in the "/usr/local/bin" directory.
 This is a good option to make "matlab" globally accessible.

--- a/docs/reference/changelog-r2020.md
+++ b/docs/reference/changelog-r2020.md
@@ -25,9 +25,10 @@ Released on XXX.
     - Fixed addition of PROTO nodes including a [Connector](connector.md) node from the Add Node dialog (thanks to Acwok).
     - Fixed the [`wb_display_image_load`](display.md#wb_display_image_load) function when used with a PNG image with transparency.
     - Fixed color of the bounding objects remaining in the collision state if the collision was lasting only one step (thanks to Acwok).
-    - Fixed Matlab API.
+    - Fixed MATLAB API.
     - Fixed missing stdout/stderr flush when a controller is changed or restarted while simulation is running (thanks to tsampazk).
     - Fixed the ROS controller of the Universal Robots UR3e, UR5e and UR10e to send the success state when a trajectory succeeded (thanks to Tim-Stoll).
+    - Fixed support for MATLAB R2017b on Windows.
 
 ## Webots R2020a Revision 1
 Released on January 14th, 2020.

--- a/lib/controller/matlab/launcher.m
+++ b/lib/controller/matlab/launcher.m
@@ -33,10 +33,14 @@ addpath([WEBOTS_HOME '/lib/controller/matlab']);
 if ispc
   setenv('MINGWROOT', strcat(WEBOTS_HOME,'\\msys64\\mingw64'));
   libname = 'Controller';
-  installed_addons = matlab.addons.installedAddons;
-  installed = sum(installed_addons.Identifier == "ML_MINGW");
-  if installed <= 0 || matlab.addons.isAddonEnabled("ML_MINGW") <= 0
-    disp('The MATLAB "MinGW-w64 C/C++ Compiler" addon is not installed, please install it from: https://fr.mathworks.com/matlabcentral/fileexchange/52848-matlab-support-for-mingw-w64-c-c-compiler');
+  % in MATLAB 2017b, the following tests are crashing MATLAB
+  % in MATLAB 2019b, the MingGW-w64 C/C++ Compiler is not needed any more
+  if version('-release') == '2018b'
+    installed_addons = matlab.addons.installedAddons;
+    installed = sum(installed_addons.Identifier == "ML_MINGW");
+    if installed <= 0 || matlab.addons.isAddonEnabled("ML_MINGW") <= 0
+      disp('The MATLAB "MinGW-w64 C/C++ Compiler" addon is not installed, please install it from: https://fr.mathworks.com/matlabcentral/fileexchange/52848-matlab-support-for-mingw-w64-c-c-compiler');
+    end
   end
   addpath([WEBOTS_HOME '/msys64/mingw64/bin']);
 else
@@ -58,11 +62,11 @@ try
   lockfile = 'webots_matlab_lock';
   counter = 1;
   while exist(lockfile, 'file')
-    if (counter == 1)
+    if counter == 1
       disp(['Waiting up to 5 seconds for ' tempdir lockfile ' to be deleted by another MATLAB instance.']);
     end
     pause(1);
-    if (counter == 5)
+    if counter == 5
       disp(['Deleting ' tempdir lockfile '...'])
       delete(lockfile);
     end
@@ -119,7 +123,7 @@ try
       disp('Load Library successful');
       libcontrollerloaded = true;
     catch ME % this happens with MATLAB R2015a only but seems to be harmless
-      if (version('-release') == '2015a')
+      if version('-release') == '2015a'
         loadlibrary(libname,protofile,'alias','libController');
         libcontrollerloaded = true;
       else

